### PR TITLE
Check kubectl version before failing immutable configmap test

### PR DESCRIPTION
**Pull request by sishbi**
## Description
Check kubectl version before failing immutable configmap test

## Issues:
Refs: #669

## How has this been tested:
 - [X] Covered by existing integration testing
 - [ ] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

Please review: @taylor @wavell

### CODE REVIEW

File: src/tasks/utils/kubectl_client.cr [hunk 1]
```diff
@@ -33,6 +33,26 @@ module KubectlClient
     LOGGING.info "KubectlClient.cp stderr: #{stderr.to_s}"
     {status: status, output: output, error: stderr}
   end
+  def self.version()
```
Comment (williscool) [inline src/tasks/utils/kubectl_client.cr:36]: can you write a test for this please?

concerned about weird/ number string stuff

File: src/tasks/workload/configuration_lifecycle.cr [hunk 2]
```diff
@@ -465,12 +465,20 @@ task "immutable_configmap" do |_, args|
...
 
+    immutable_configmap_supported = true
     # if the reapply with a change succedes immmutable configmaps is NOT enabled
     # if KubectlClient::Apply.file(test_config_map_filename) == 0
     if KubectlClient::Apply.file(test_config_map_filename)
       LOGGING.info "kubectl apply failed for: #{test_config_map_filename}"
-      resp = "✖️  FAILED: immmutable configmaps are not enabled in this k8s cluster.".colorize(:red)
-      upsert_failed_task("immutable_configmap", resp)
+      k8s_ver = KubectlClient.version
+      if k8s_ver <= "1.18"
```
Comment (williscool) [inline src/tasks/workload/configuration_lifecycle.cr:474]: does this work with a string / number?

**Review by williscool — CHANGES_REQUESTED** [review]

File: src/tasks/workload/configuration_lifecycle.cr (hunk 2 above)
Comment (taylor) [reply src/tasks/workload/configuration_lifecycle.cr:474]: https://github.com/crystal-lang/crystal/blob/master/src/semantic_version.cr

File: src/tasks/utils/kubectl_client.cr (hunk 1 above)
Comment (sishbi) [reply src/tasks/utils/kubectl_client.cr]: Does using semantic version library resolve this concern?

File: src/tasks/workload/configuration_lifecycle.cr (hunk 2 above)
Comment (sishbi) [reply src/tasks/workload/configuration_lifecycle.cr]: I have implemented the check using semantic_version library

**Pull request merged**